mjit_c.rb

```diff
@@ -439,32 +439,6 @@ def C.rb_cref_t
   def C.rb_execution_context_struct
     @rb_execution_context_struct ||= CType::Struct.new(
       "rb_execution_context_struct", Primitive.cexpr!("SIZEOF(struct rb_execution_context_struct)"),
-      vm_stack: [CType::Pointer.new { self.VALUE }, Primitive.cexpr!("OFFSETOF((*((struct rb_execution_context_struct *)NULL)), vm_stack)")],
```

Comment:
Why was this file changed?

Reply:
I applied the diff from https://github.com/Shopify/ruby/actions/runs/4313674348/jobs/7525759736 in order to get the build passing again.

I don't know much about mjit, or the bindgen code - but the `rb_execution_context_struct` has changed (to extract the shape tree), but it hasn't changed this much.

Reply:
This turned out to be a bug in the `mjit`/`rjit` bindgen code - when we're generating the bindings we grab all the nodes of the same type - we can have multiple nodes of the same type if there are multiple declarations in different header files, as there are with `rb_execution_context_struct` - but the code that generates the bindings uses the last node instance to find the struct members.

It looks like my code has changed the order in which the nodes are discovered so that the last entry is now just an empty declaration, instead of the actual declaration, and so rjit bindgen thinks it has no members.

Fixed by @k0kubun in https://github.com/ruby/ruby/commit/309dd50a01203ac107c12e5a962e6aa6add75529 so that now we search the node list for the declaration that contains the member definitions.